### PR TITLE
[Fix for # 810]Porting few tests for JsonOutputFormatter

### PR DESCRIPTION
**Pull request by harshgMSFT**
#810

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.Core.Test/Formatters/JsonOutputFormatterTests.cs [hunk 1]
```diff
@@ -87,8 +89,100 @@ public void Creates_SerializerSettings_ByDefault()
...
-        private OutputFormatterContext GetOutputFormatterContext(object outputValue, Type outputType,
-                                            string contentType = "application/xml; charset=utf-8")
+        [Fact]
+        public async Task WriteToStreamAsync_RoundTripsJToken()
+        {
+            var beforeMessage = "Hello World";
+            var formatter = new JsonOutputFormatter();
+            JToken before = new JValue(beforeMessage);
+            MemoryStream memStream = new MemoryStream();
+            var outputFormatterContext = GetOutputFormatterContext(
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/Formatters/JsonOutputFormatterTests.cs]: vars

Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/Formatters/JsonOutputFormatterTests.cs]: `// Arrange // Act // Assert`

File: test/Microsoft.AspNet.Mvc.Core.Test/Formatters/JsonOutputFormatterTests.cs [hunk 2]
```diff
@@ -87,8 +89,100 @@ public void Creates_SerializerSettings_ByDefault()
...
+
+            // Assert
+            var actualData = memStream.ToArray();
+            Assert.Equal(expectedData, actualData);
+        }
+
+        private static Encoding CreateOrGetSupportedEncoding(
+            JsonOutputFormatter formatter, string encoding, bool isDefaultEncoding)
+        {
+            Encoding enc = null;
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/Formatters/JsonOutputFormatterTests.cs]: don't abbreviate

File: test/Microsoft.AspNet.Mvc.Core.Test/Formatters/JsonOutputFormatterTests.cs (hunk 1 above)
Comment (harshgMSFT) [inline test/Microsoft.AspNet.Mvc.Core.Test/Formatters/JsonOutputFormatterTests.cs]: done.

File: test/Microsoft.AspNet.Mvc.Core.Test/Formatters/JsonOutputFormatterTests.cs (hunk 2 above)
Comment (harshgMSFT) [inline test/Microsoft.AspNet.Mvc.Core.Test/Formatters/JsonOutputFormatterTests.cs]: fixed.

**Comment by rynowak** [comment]
Do a quick pass here for style issues, this is sloppy compared to the new code we're writing :watch:

**Comment by harshgMSFT** [comment]
updated

File: test/Microsoft.AspNet.Mvc.Core.Test/Formatters/JsonOutputFormatterTests.cs [hunk 3]
```diff
@@ -109,7 +212,7 @@ private static ActionContext GetActionContext(MediaTypeHeaderValue contentType)
             request.SetupGet(r => r.Headers).Returns(headers);
             headers[HeaderNames.AcceptCharset] = contentType.Charset;
             var response = new Mock<HttpResponse>();
-            response.SetupGet(f => f.Body).Returns(new MemoryStream());
+            response.SetupGet(f => f.Body).Returns(responseStream ?? new MemoryStream());
             var httpContext = new Mock<HttpContext>();
             httpContext.SetupGet(c => c.Request).Returns(request.Object);
             httpContext.SetupGet(c => c.Response).Returns(response.Object);
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/Formatters/JsonOutputFormatterTests.cs]: nit: you can simply most of this by just using DefaultHttpContext

**Comment by rynowak** [comment]
:shipit:

**Pull request closed without merging**